packages/xrpl/test/models/depositPreauth.test.ts

```diff
@@ -108,4 +157,154 @@ describe('DepositPreauth', function () {
       "DepositPreauth: Account can't unauthorize its own address",
     )
   })
+
+  it('throws when AuthorizeCredentials is not an array', function () {
+    const errorMessage = 'DepositPreauth: Credentials list must be an array'
+    depositPreauth.AuthorizeCredentials = validCredential
+
+    assert.throws(
+      () => validateDepositPreauth(depositPreauth),
+      ValidationError,
+      errorMessage,
+    )
+    assert.throws(() => validate(depositPreauth), ValidationError, errorMessage)
+  })
+
+  it('throws when UnauthorizeCredentials is not an array', function () {
+    const errorMessage = 'DepositPreauth: Credentials list must be an array'
+    depositPreauth.UnauthorizeCredentials = validCredential
+
+    assert.throws(
+      () => validateDepositPreauth(depositPreauth),
+      ValidationError,
+      errorMessage,
+    )
+    assert.throws(() => validate(depositPreauth), ValidationError, errorMessage)
+  })
+
+  it('throws when AuthorizeCredentials is empty array', function () {
+    const errorMessage = 'DepositPreauth: Credentials list cannot be empty'
```

Comment:
It's more explicit stating an array than a list.
```suggestion
    const errorMessage = 'DepositPreauth: Credentials cannot be an empty array'
```